enhancements/monitoring/integration-with-topology-modes.md

```diff
@@ -0,0 +1,176 @@
+redundancy in term of metrics scraping, rule evaluations and querying as long
+as one of the Prometheus instances is still up.
+
+The Thanos ruler instances of user workload monitoring work the same as
+Prometheus except they only deal with rule evaluations.
+
+The 3 Alertmanager instances receive alerts from both Prometheus instances and
+as long as one instance is up and running, the notifications should flow to the
+receivers. Alertmanager instances also replicate silences and notification logs
+between themselves to prevent duplicate notifications.
+
+The other monitoring components running with multiple replicas in HA mode
+(thanos-querier and prometheus-adapter) are stateless. Running multiple
+replicas only ensures that their services remain available as long as 1
+instance is up.
+
+By definition, the monitoring stack wouldn't be highly-available in the
+`SingleReplica` mode. One recommendation would be that cluster admins configure
+Prometheus and Alertmanager with persistent storage to ensure that existing
+metrics, silences and notification logs are retained upon pod recreation.
+
+## Design Details
+
+### Test Plan
+
+We will setup a CI job that runs the `extended` end-to-end test suite from
+[OpenShift Origin](https://github.com/openshift/origin) in a single-node
+deployment mode. The test suite has already been extended to verify that
+operators honor the `SingleReplica` value (see
+[openshift/origin#25812](https://github.com/openshift/origin/pull/25812) and
+[openshift/origin/pull/25885](https://github.com/openshift/origin/pull/25885)).
+
+As soon as the implementation in CMO is merged, we will adjust the single-node
+topology
+[tests](https://github.com/openshift/origin/blob/master/test/extended/single_node/topology.go)
+to remove the exception in place for the monitoring components.
+
+Cluster admins can use the `openshift-monitoring/cluster-monitoring-operator`
+configmap to tune the configuration of the operator and its operands (
+user-workload monitoring enablement, persistent storage, resource
+requests/limits, ...). We have [end-to-end tests](https://github.com/openshift/cluster-monitoring-operator/tree/master/test/e2e)
+in the cluster-monitoring-operator verifying that these configuration
+parameters work as expected. Because the CMO configuration doesn't allow to
+change the replica count and it doesn't expose the topology mode either, we
+don't plan to add any test specific to the single-node mode to the CMO test
+suite.
+
+### Graduation Criteria
+
+N/A
+
+### Upgrade / Downgrade Strategy
+
+Upgrades are a non goal right now.
+
+### Version Skew Strategy
+
+If CMO reads a value that is neither equal to `HighAvailable` nor
+`SingleReplica`, it should consider that the topology is `HighAvailable`.
+
+## Implementation History
+
+## Drawbacks
+
+N/A
+
+## Alternatives
+
+* Do nothing. CMO would deploy multiple replicas for Prometheus Alertmanager,
+* Thanos querier, Thanos ruler and prometheus-adapter. As a result, the
```

Comment:
The formatting here looks suspicious. I think lines 170-171 shouldn't start with `*`.